gems/aws-partitions/lib/aws-partitions/endpoint_provider.rb

```diff
@@ -54,27 +60,50 @@ def signing_region(region, service)
           .fetch('region', region)
       end
 
+      # @param [String] region The region used to fetch the partition.
+      # @param [String] service Used only if dualstack is true. Used to find a
+      #   DNS suffix for a specific service.
+      # @param [Boolean] dualstack Set to `true` to use a dualstack DNS suffix
+      #   if available. Falls back to normal suffixes.
       # @api private Use the static class methods instead.
-      def dns_suffix_for(region)
-        get_partition(region)['dnsSuffix']
+      def dns_suffix_for(region, service, dualstack, fips)
```

Comment:
we may want to make the dualstack/fips generic to variants.  Maybe a variants hash